controllers/auth_config_status_updater.go

```diff
@@ -41,41 +46,152 @@ func (u *AuthConfigStatusUpdater) Reconcile(ctx context.Context, req ctrl.Reques
 	} else {
 		// resource found and it is to be watched by this controller
 		// we need to update its status
-		if err := u.updateAuthConfigStatus(ctx, req.String(), &authConfig, true); err != nil {
-			logger.Info(err.Error())
+		if err := u.updateAuthConfigStatus(log.IntoContext(ctx, logger), req.String(), &authConfig); err != nil {
 			return ctrl.Result{Requeue: true}, nil
 		} else {
-			logger.Info("resource status updated")
 			return ctrl.Result{}, nil
 		}
 	}
 }
 
-func (u *AuthConfigStatusUpdater) updateAuthConfigStatus(ctx context.Context, cacheId string, authConfig *api.AuthConfig, ready bool) error {
-	if len(u.Cache.FindKeys(cacheId)) == 0 {
-		return fmt.Errorf("resource not ready")
+func (u *AuthConfigStatusUpdater) updateAuthConfigStatus(ctx context.Context, cacheId string, authConfig *api.AuthConfig) (err error) {
+	logger := log.FromContext(ctx)
+
+	var condition api.Condition
+	linked, loose := u.partitionHostsByStatus(cacheId, authConfig)
+	if len(loose) > 0 {
+		condition = statusNotReady(api.StatusReasonHostNotLinked, "One or more hosts not linked to the resource")
+	} else {
+		condition = statusReady()
 	}
 
-	authConfig.Status.Ready = ready
-	authConfig.Status.NumIdentitySources = int64(len(authConfig.Spec.Identity))
-	authConfig.Status.NumMetadataSources = int64(len(authConfig.Spec.Metadata))
-	authConfig.Status.NumAuthorizationPolicies = int64(len(authConfig.Spec.Authorization))
-	authConfig.Status.NumResponseItems = int64(len(authConfig.Spec.Response))
+	var conditionsChanged, summaryChanged bool
 
-	issuingWristbands := false
-	for _, responseConfig := range authConfig.Spec.Response {
-		if responseConfig.GetType() == api.ResponseWristband {
-			issuingWristbands = true
-			break
-		}
+	authConfig.Status.Conditions, conditionsChanged = buildStatusConditions(authConfig.Status.Conditions, condition)
+	authConfig.Status.Summary, summaryChanged = buildStatusSummary(authConfig, linked)
+
+	if !authConfig.Status.Ready() {
+		err = fmt.Errorf("resource not ready")
 	}
-	authConfig.Status.FestivalWristbandEnabled = issuingWristbands
 
-	return u.Status().Update(ctx, authConfig)
+	if changed := conditionsChanged || summaryChanged; !changed {
+		logger.V(1).Info("resource status did not change")
+		return // to save an update request
+	}
+
+	logger.V(1).Info("resource status changed", "authconfig/status", authConfig.Status)
+
+	if updateErr := u.Status().Update(ctx, authConfig); updateErr != nil {
+		logger.Error(updateErr, "failed to update the resource")
+		err = updateErr
+	}
+
+	logger.Info("resource status updated")
+
+	return
+}
+
+func (u *AuthConfigStatusUpdater) partitionHostsByStatus(cacheId string, authConfig *api.AuthConfig) ([]string, []string) {
+	type obj struct{}
+	hosts := make(map[string]obj)
+	for _, host := range u.Cache.FindKeys(cacheId) {
+		hosts[host] = obj{}
+	}
+	linked := []string{}
+	loose := []string{}
```

Comment:
`var linked, loose []string`

Reply:
It raises an error in cases where no hosts are linked, therefore nothing is ever appended to `linked`, and later `linked` is used to populate `summary.HostsReady`. Since the field does not accept null values, of fails. That's why I default to initialising the slices always.